### PR TITLE
Platform Engineering Maturity Model v1

### PR DESCRIPTION
**Pull request by joshgav**
This is the culmination of WG Platforms' last few months of effort to provide a next step for readers of the Platforms White Paper. We'll merge this to main to finalize publication and represent final acceptance by the TAG and CNCF TOC.

### CODE REVIEW

**Comment by joshgav** [comment]
@cjyabraham can you please force the Netlify site to render for this PR? In fact it would be great if from now on it rendered for everything even if outside of the `website` dir - could you do that now? Thanks!

**Comment by cjyabraham** [comment]
Looks like my attempt to always force builds has worked. [Here is the preview](https://deploy-preview-472--tag-app-delivery.netlify.app/). 

I'll close out [the PR I created for this](https://github.com/cncf/tag-app-delivery/pull/473) as the change is now included in this PR.

**Review by roberthstrand — APPROVED** [review]
Awesome job to everyone involved. This looks **great**!

**Comment by joshgav** [comment]
Thank you @cjyabraham, looks great and already very helpful in discovering issues!

Changes to the base branch for this PR will be a perfect way to test your changes :wink:, as I understand they should each trigger a redeploy.

File: platforms-maturity-model/v1/index.md
Comment (blakeromano) [inline platforms-maturity-model/v1/index.md:115]: Do Tiger teams always tackle just technical requirements? Or Product requirements as well?

File: platforms-maturity-model/v1/index.md
Comment (blakeromano) [inline platforms-maturity-model/v1/index.md:127]: This wording does seem to lean to very internal cloud platforms. May want to think about to how word it to be more generic.

File: platforms-maturity-model/v1/index.md
Comment (blakeromano) [inline platforms-maturity-model/v1/index.md:193]: I'd remove DevOps in this paragraph in order to keep it more for generic platforms.

**Review by blakeromano — COMMENTED** [review]
Couple comments relating to some of the wording being very closely coupled with Internal Cloud Platforms that I think may benefit from getting re-worded to be more generic.

File: platforms-maturity-model/v1/index.md
Comment (joshgav) [reply platforms-maturity-model/v1/index.md]: Fixed

**Comment by joshgav** [comment]
@cjyabraham can you please trigger a rebuild following the latest PR merge? Thanks.

**Comment by cjyabraham** [comment]
> @cjyabraham can you please trigger a rebuild following the latest PR merge? Thanks.

Which PR merge are you referring to?

File: platforms-wg/glossary/_index.md
Comment (abangser) [reply platforms-wg/glossary/_index.md:10]: I definitely see what you mean. I wonder if we need an intro like below would help?

> A platform combines processes and technologies in order to support the building of higher level applications, processes or technologies. In the context of this working group and related publications, the focus is on platforms that support the  development and delivery of products, services and apps.

WDYT? Cc/@joshgav

File: platforms-maturity-model/v1/index.md
Comment (blakeromano) [reply platforms-maturity-model/v1/index.md:115]: I think removing `technical` would make more sense. I like that concept of a migration because a migration could be technical (migration from old software to new to improve technical) but you could also think of it as a tiger team migrating from one software to another to remove the need for a contract which would not be a technical requirement but maybe a business/cost requirement

File: platforms-maturity-model/v1/index.md
Comment (blakeromano) [reply platforms-maturity-model/v1/index.md:127]: I think really

>  They may be called DevOps, Engineering Enablement, Developer Experience (DevEx or DevX), Shared Tools, a Centre-Of-Excellence, or even Platform.

Is the part that concerns me. Maybe adding the disclaimer at the top that some of this leans towards app delivery platforms may be good enough to warrant this sentence being in there.

**Comment by krumware** [comment]
Personally, seeing it live, I think it's a little bit funky to have the Context tab alongside the other aspects. It feels like something that needs to be separated or perhaps precede that functional section of tabs. Placing the Context content before the aspects table seems like an appropriate place and could give the web content some needed length.

Separately, those tabs are going to be difficult for some users to notice at first, so a quick instruction statement or tooltip prior to the tabs section might be nice.

Quick note, the target hashes in the links in the aspects table (first column), don't match the tab targets (which look like random chars)

**Comment by Saim-Safdar** [comment]
> Personally, seeing it live, I think it's a little bit funky to have the Context tab alongside the other aspects. It feels like something that needs to be separated or perhaps precede that functional section of tabs. Placing the Context content before the aspects table seems like an appropriate place and could give the web content some needed length.
> 
> Separately thought, those tabs are going to be difficult for some users to notice at first so a quick instruction statement or tooltip prior to the tabs section might be nice.
> 
> Last note, the target hashes in the links in the aspects table (first column), don't match the tab targets (which look like random chars)

I feel like the same,  tabs should look different than the links, at first look the same, but super difficult to differentiate at the very first notice.

File: platforms-maturity-model/v1/index.md
Comment (abangser) [reply platforms-maturity-model/v1/index.md:120]: 👍 That makes sense to me, incoming change to remove technical!

File: platforms-wg/glossary/_index.md
Comment (abangser) [reply platforms-wg/glossary/_index.md:10]: This sections changed a bit with Josh's changes so I have incorporated the intention of my edit in a new commit 👍

File: platforms-maturity-model/v1/index.md
Comment (abangser) [reply platforms-maturity-model/v1/index.md:132]: This has been added into the intro around the paper so the caveat about this group (and therefore this paper) focus is included.

Thanks for the suggestion!

**Comment by AdminTurnedDevOps** [comment]
@joshgav @roberthstrand 

Hey All - I don't know if you're still accepting requests in terms of some edits, but I wanted to run something by you in terms of the overall definition of Platform Engineering.

I came up with the following definition:

"Platform Engineering is a list of integrated capabilities that are created according to what the platform user needs (the platform user is the internal engineer/developer using the platform)."

From what I've seen, it's really the ability to create internal systems for internal use. It's like the employees working in other engineering teams are your customers, and they should almost be treated as such from a product perspective... that product being the internal platform you're creating to make processes more repeatable for them.

Just my two cents! Please let me know if this sounds good and I'm happy to answer any questions.

**Comment by AdminTurnedDevOps** [comment]
Adding another comment/extension to my comment from two days ago.

I did extensive research this weekend in all avenues of forums to comments to articles to really nail down exactly how people are viewing Platform Engineering right now and it turns out that the majority of people aren’t 100% sure what it is. There’s still a lot of confusion around it. The majority of the confusion comes from wondering what the differences are between DevOps, SRE, and the other avenues of engineering out there.

I believe the best differentiation is: Platform Engineering provides a focus for building internal tools (like self-service tools) to make the lives of internal engineers easier. They have a self-service method (a UI, CLI, or some automation/repeatable process) of creating the exact environment they need in a way that they don’t need to be bogged down by the details of how it’s done. It’s treating internal engineering as a product within itself.

**Comment by abangser** [comment]
@AdminTurnedDevOps These are fantastic points and absolutely love that you are identifying the needs in the community and turning them into opportunities for this group to help give clarification!

I believe that the points you have raised are best suited to a revamp on the Glossary which I believe we are going to undertake post KubeCon taking on board what we have heard recently on group calls preparing this paper, your comments, and anything we can gather while at the event in Chicago. I have added you as a cc on the issue here: https://github.com/cncf/tag-app-delivery/issues/485

**Comment by AdminTurnedDevOps** [comment]
@abangser Thank you! I appreciate it :).

**Comment by joshgav** [comment]
Code owners PTAL, we need your approval to merge today:
- @roberthstrand 
- @AloisReitbauer 
- @AlexsJones 
- @lianmakesthings 
- @kgamanji 
- @justincormack 
- @angellk

**Review by angellk — APPROVED** [review]
lgtm - appreciate all the hard work everyone put into this!

**Pull request merged**
